### PR TITLE
Audit triplestore

### PR DESCRIPTION
**Pull request by mohideen**
Functional workflow: Camel picks up JMS events from fedora and updates Fuseki triplestore.

### CODE REVIEW

**Comment by awoods** [comment]
I assume `mvn clean install` fails for you, @mohideen, as well?

**Comment by mohideen** [comment]
Yes, I have been skipping tests for quick checks. Missed to run the tests after finishing the work. Will update with fixes for the tests.

File: audit-triplestore/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 1]
```diff
@@ -12,7 +12,11 @@
      <cm:default-properties>
        <cm:property name="jms.brokerUrl" value="tcp://localhost:61616"/>
        <cm:property name="jms.fcrepoEndpoint" value="topic:fedora"/>
-       <cm:property name="triplestore.baseUrl" value="localhost:3030/ds/test"/>
+       <cm:property name="triplestore.baseUrl" value="localhost:3030/fedora4"/>
```
Comment (escowles) [inline audit-triplestore/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: Is there any benefit to having two properties?  Why not just make triplestore.baseUrl localhost:3030/fedora4/update?

File: audit-triplestore/src/main/java/org/fcrepo/camel/audit/triplestore/AuditSparqlProcessor.java [hunk 2]
```diff
@@ -114,22 +116,24 @@ public void process(final Exchange exchange) throws Exception {
     private static final String NODE_REMOVED = REPOSITORY + "NODE_REMOVED";
     private static final String PROPERTY_CHANGED = REPOSITORY + "PROPERTY_CHANGED";
 
+    private static final String EMPTY_STRING = "";
+
     /**
      * Convert a Camel message to audit event description.
-     * @param m JMS message produced by an audit event
+     * @param message JMS message produced by an audit event
```
Comment (escowles) [inline audit-triplestore/src/main/java/org/fcrepo/camel/audit/triplestore/AuditSparqlProcessor.java]: I thought I had fixed this already, but this should say it's a Camel message, not JMS.

File: audit-triplestore/src/main/resources/OSGI-INF/blueprint/blueprint.xml (hunk 1 above)
Comment (acoburn) [inline audit-triplestore/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: Agreed -- this should be a single property

File: audit-triplestore/src/test/java/org/fcrepo/camel/audit/triplestore/RouteTest.java [hunk 3]
```diff
@@ -59,7 +59,7 @@ public void testEventProcessor() throws Exception {
         final Map<String, Object> headers = new HashMap<>();
         headers.put(JmsHeaders.BASE_URL, "http://localhost/rest");
         headers.put(JmsHeaders.IDENTIFIER, "/foo");
-        headers.put(JmsHeaders.TIMESTAMP, "1428360320168");
+        headers.put(JmsHeaders.TIMESTAMP, 1428360320168L);
```
Comment (acoburn) [inline audit-triplestore/src/test/java/org/fcrepo/camel/audit/triplestore/RouteTest.java]: Does the JMS message from fedora have this value serialized as a Long? Correct me if I'm wrong, but I thought everything comes across the wire as a String.

Comment (escowles) [inline audit-triplestore/src/test/java/org/fcrepo/camel/audit/triplestore/RouteTest.java]: I was wondering that too, but it looks like fcrepo4 is setting it as a long:

https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-jms/src/main/java/org/fcrepo/jms/headers/DefaultMessageFactory.java#L79

**Comment by escowles** [comment]
:+1 this looks good to me & is working for me locally.

**Comment by escowles** [comment]
Wait, looks like there's a merge conflict in the RouterTest -- can you rebase this?

**Comment by mohideen** [comment]
Sure, I will merge fcrepo-camel-toolkit:master to PR branch.

**Comment by escowles** [comment]
:+1: now it merges cleanly too – thanks @mohideen

File: audit-triplestore/src/main/java/org/fcrepo/camel/audit/triplestore/AuditSparqlProcessor.java [hunk 4]
```diff
@@ -62,7 +61,9 @@
      */
     public void process(final Exchange exchange) throws Exception {
         final Message in = exchange.getIn();
-        final UriRef eventURI = new UriRef("XXX");
+        final String eventURIBase = (String) exchange.getProperty("eventURI.base");
```
Comment (awoods) [inline audit-triplestore/src/main/java/org/fcrepo/camel/audit/triplestore/AuditSparqlProcessor.java]: Does a mixed-case, String-literal property not seem risky?

Comment (acoburn) [inline audit-triplestore/src/main/java/org/fcrepo/camel/audit/triplestore/AuditSparqlProcessor.java]: It's not particularly risky given that it's set in the EventRouter class from a property placeholder value. But wouldn't it make more sense to put the value in a header? (I realize that the distinction b/t header and property is a bit thin)

Comment (awoods) [inline audit-triplestore/src/main/java/org/fcrepo/camel/audit/triplestore/AuditSparqlProcessor.java]: I will leave the question of `property` vs. `header` up to you...
We can certainly leave `eventURI.base` as it is; however, I notice an existing convention along the lines of:
- `triplestore.baseUrl`

Close, but different.

Comment (acoburn) [inline audit-triplestore/src/main/java/org/fcrepo/camel/audit/triplestore/AuditSparqlProcessor.java]: yes, the property names should follow a more consistent naming pattern

**Pull request merged**